src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import core.basesyntax.service.ValidationException;
+import java.util.Random;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationServiceImpl registrationService;
+    private User user = new User();
+    private Random random = new Random();
+    private String login = "admin" + random.nextInt(6, 99999);
+    private String password = "admin" + random.nextInt(6, 99999);
+    private int userAge = random.nextInt(18, 99);
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_auth_ok() {
+        user.setLogin(login);
+        user.setPassword(password);
+        user.setAge(userAge);
+
+        assertEquals(registrationService.register(user), user);
+
+        User user2 = new User();
+        user2.setLogin("tre" + random.nextInt(9) + random.nextInt(9) + random.nextInt(9));
+        user2.setPassword("123456");
+        user2.setAge(18);
+        assertEquals(registrationService.register(user2), user2);
+
+    }
+
+    @Test
+    public void register_auth_notok() {
+        user.setLogin("five5");
+        user.setPassword("five");
+        user.setAge(17);
+
+        assertThrows(ValidationException.class, () -> registrationService.register(user));
+
+        User user2 = new User();
+        user2.setLogin("");
+        user2.setPassword("");
+        user2.setAge(0);
+
+        assertThrows(ValidationException.class, () -> registrationService.register(user2));
+
+    }
+
+    @Test
+    public void register_setidafterauth_ok() {
```

Comment:
There is no any logic about setting id in register method, so please remove this test case